### DISCUSSION TITLE
Spacing vs Extrusion Width on first layer

### DISCUSSION THREAD

**Discussion by UD551** [supermerill/SuperSlicer#2211] [Q&A] [answered]
Hello all,
I am having trouble distinguishing between what is meant by "spacing" and "extrusion width" on the first layer.  I  looked at the explanation help/details tab, but I couldn't understand the concept. In theory, isn't spacing< extrusion width? 
In addition, how does having an overlap at 0% lead to spacing = extrusion?  In this case, when referring to the overlap, you mean "infill perimeter overlap"? If so, how will the infill perimeter overlap affect the spacing between the solid infill?
Basically, I am trying to figure out what is the actual filament width (excluding the seam) after the filament has been printed and bonded to each other. 

Another concern I have is when changing the first layer flow ratio. Does changing the first layer flow ratio affect the spacing between the filaments? or is the spacing constant whatever the first layer flow ratio is?  If this is the case, then what affect will the increase in flow ratio have on the first layer? i.e. where will the extra flow of filament go? Does the software compensate for such increase in flow by increasing the programmed layer height in order to keep the specified spacing constant? 

I hope I was able to convey  my concern properly. Thank you so much for the help.  Please let me know if need to re-explain my concern. 
Keep up the great work!

**Comment by supermerill** [comment, marked as answer]
> , isn't spacing< extrusion width?

yes

> how does having an overlap at 0% lead to spacing = extrusion?

no overlap => extrusion aren't touching (or at a point) => same spacing as width

>  you mean "infill perimeter overlap"?

no, to avoid this confusion, I renamed it "infill perimeter encroaching" (but it's in the dev build only)

>  how will the infill perimeter overlap affect the spacing between the solid infill?

it doesn't

> what is the actual filament width (excluding the seam) after the filament has been printed and bonded to each other.

Nobody know for sure. The fluid physic is capricious, and depending on your nfilament type, brand, heat, speed... it can have a bit different shape irl.

If you extrude in vase mode, the theoritical width of the extrusion as measured by a caliper is the 'external perimeter width', but i think it's a bit more irl as the extrusion seems more "pointy" than the ideal half-circle we use.
If you extrude a top solid infill, then as there is overlap between line to have a full fill without gaps, "filament width" doesn't have a real meaning.

**Reply by supermerill** [reply]
> Does changing the first layer flow ratio affect the spacing between the filaments?

no. And i advise to not use it unless your hardware is crappy.

>  is the spacing constant whatever the first layer flow ratio is?

If you modify the flow, then the other computations doesn't have meaning anymore as you're throwing their computation in the bath by saying "whatever, just push X% more, it's crappy anyway"

>  then what affect will the increase in flow ratio have on the first layer?

if you set 120%, then your first layer will have 20% more plastic, and so weight 20% more than with  100%.

> where will the extra flow of filament go?

where it can, or your extruder skip steps

> Does the software compensate for such increase in flow by increasing the programmed layer height in order to keep the specified spacing constant?

no

**Reply by UD551** [reply]
Thank you so much for your quick response! I really appreciate it. 
Concerning the "infill perimeter encroaching", how can I get access/download the dev build file? Theoretically, will I be able to eliminate the gaps between the solid infill on the first layer if I use an infill perimeter encroaching of 100%? 

On the most recent super slicer version, what can I do to eliminate the gaps between the solid infill on the first layer? What settings can I manipulate to make sure the gaps are filled? 
With that said, shouldn't the spacing value change as the %infill perimeter encroaching between the solid infill change? If that's the case then, the spacing value is **not changing** when I change the infill perimeter encroaching nor the perimeter overlap on the newest super slicer version. The only setting that is changing the value of the spacing is the base layer height. 
![Overlap _Spacing](https://user-images.githubusercontent.com/87382555/149235828-bd666c7e-a727-43ed-972b-6604f4b3da22.jpg)

Thank you for your time! I hope the image I attached explains my issue properly.

**Reply by supermerill** [reply]
It's just a rename, nothing changed.
it's not linked. The encroachment just put a bit on the infill area inside the perimeter area to be sure there is a good bonding, that's all.
If your first layer is under-extruded, maybe your bed is too far away from your nozzle. there is a calibration for that.

**Reply by DocRuzzy** [reply]
what formula is used to calculate the extrusion width if inputs are
- set spacing (0.8)
- base layer height (0.3)
- perimeter overlap (100%)
- First layer flow ratio (100%)
?

With these values the indicated EW = 0.8644. It doesn't seem to change if we set perimeter overlap = 0%.

Is it true that if we set the perimeter overlap = 0% , then Extrusion width = line spacing?  (exactly as we tuned our extrusion multiplier)
If we then use a flow ratio modifier (First layer) will that modify the flow rate to achieve an EW defined by the line spacing?

The issue is, even though we have very precisely tuned our z-offset and extrusion multiplier, using a perimeter overlap of 100% is NOT enough to remove the gaps on the first layer (we can see them on the bottom).

We have been able to close them by changing the first layer flow ratio, but we'd like to understand what that First layer flow ratio is multiplying. (i.e. perimeter overlap included at all, is FL height used or base layer height).

I know this might seem tedious but we very much appreciate your correspondence.

**Reply by supermerill** [reply]
> Is it true that if we set the perimeter overlap = 0% , then **perimeter** Extrusion width = **perimeter** line spacing? 

yes

> If we then use a flow ratio modifier (First layer) will that modify the flow rate to achieve an Extrusion Width defined by the line spacing?

If you use a flow modifier, then the flow will be changed. If it's higher than 100%, then there will be more filament than what's shown in the preview, and the extra plastic will go somewhere.

> we'd like to understand what that First layer flow ratio is multiplying.

The E value in the gcode, only on the first (object) layer.

**Reply by DocRuzzy** [reply]
Thank you!
